specification/cognitiveservices/data-plane/Language/preview/2021-11-01-preview/textanalytics.json

```diff
@@ -0,0 +1,2106 @@
+                        "sentimentAnalysisTasks": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/SentimentAnalysisTask"
+                            }
+                        },
+                        "extractiveSummarizationTasks": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/ExtractiveSummarizationTask"
+                            }
+                        }
+                    },
+                    "type": "object"
+                }
+            },
+            "required": [
+                "tasks"
+            ],
+            "type": "object"
+        },
+        "Task": {
+            "type": "object",
+            "description": "Base task object.",
+            "properties": {
+                "taskName": {
+                    "type": "string"
+                }
+            }
+        },
+        "TaskParameters": {
+            "type": "object",
+            "description": "Base parameters object for a text analysis task.",
+            "properties": {
+                "loggingOptOut": {
+                    "type": "boolean",
+                    "default": false
+                }
+            }
+        },
+        "PreBuiltTaskParameters": {
+            "type": "object",
+            "description": "Parameters object for a text analysis task using pre-built models.",
+            "properties": {
+                "modelVersion": {
+                    "type": "string",
+                    "default": "latest"
+                }
+            }
+        },
+        "PreBuiltResult": {
+            "properties": {
+                "errors": {
+                    "type": "array",
+                    "description": "Errors by record id.",
+                    "items": {
+                        "$ref": "#/definitions/RecordError"
+                    }
+                },
+                "statistics": {
+                    "$ref": "#/definitions/RequestStatistics"
+                },
+                "modelVersion": {
+                    "type": "string",
+                    "description": "This field indicates which model is used for scoring."
+                }
+            },
+            "required": [
+                "errors",
+                "modelVersion"
+            ]
+        },
+        "CustomTaskParameters": {
+            "type": "object",
+            "description": "Parameters object for a text analysis task using custom models.",
+            "properties": {
+                "project-name": {
+                    "type": "string"
+                },
+                "deployment-name": {
+                    "type": "string"
+                }
+            },
+            "required": [
+                "project-name",
+                "deployment-name"
+            ]
+        },
+        "CustomResult": {
```

Comment:
Might be good to rename this slightly to `CustomResultsEnvelope`.
There's also potential to move errors and statistics to a different `BaseResultsEnvelope` since the only difference between custom and pre-trained is on the extra-parameters.